pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -156,14 +155,18 @@ private PluginClassLoader createClassLoader(Collection<URL> urlList) {
    * @param <T>
    * @return
    */
-  public <T> T createInstance(String pluginName, String className, Class[] argTypes, Object[] argValues)
-      throws Exception {
+  public <T> T createInstance(String pluginName, String className, Class[] argTypes, Object[] argValues) throws Exception{
     PluginClassLoader pluginClassLoader = PLUGIN_MANAGER._registry.get(new Plugin(pluginName));
-    Class<T> loadedClass = (Class<T>) pluginClassLoader.loadClass(className, true);
-    Constructor<?> constructor = loadedClass.getConstructor(argTypes);
-    if (constructor != null) {
-      Object instance = constructor.newInstance(argValues);
-      return (T) instance;
+    try {
+      Class<T> loadedClass = (Class<T>) pluginClassLoader.loadClass(className, true);
+      Constructor<?> constructor = null;
+      constructor = loadedClass.getConstructor(argTypes);
+      if (constructor != null) {
+        Object instance = constructor.newInstance(argValues);
+        return (T) instance;
+      }
+    } catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException | ClassNotFoundException e) {
```

Comment:
Please add a log here with the plugin name and class name

Reply:
Isn't it better for the caller to log it?

Reply:
Maybe, maybe not. We depend on each caller to log all details needed to catch this. Sometimes it is better to just log all info at the lowest level. Many exceptions do not have strings that indicate the parameters on which the operation was being attempted. I suggesgt we log one here.